### PR TITLE
Enhancements in golang client library.

### PR DESCRIPTION
**Pull request by art2ip**
- Support for connection pooling.
- Added API for client to pass in a func to get TLS config.
- Cleaned up API.
- Added pkg/client/README.md

### CODE REVIEW

File: test/cli/server.crt
Comment (NeetishPathak) [inline test/cli/server.crt:1]: file to be removed

File: test/cli/server.pem
Comment (NeetishPathak) [inline test/cli/server.pem:1]: file to be removed

File: test/cli/server.crt
Comment (art2ip) [reply test/cli/server.crt:1]: done.

File: test/cli/server.pem
Comment (art2ip) [reply test/cli/server.pem:1]: done

**Comment by art2ip** [comment]
closed

**Pull request closed without merging**